pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -109,18 +107,21 @@ func (e *PluginManager) AddObjectMetadata(taskCtx pluginsCore.TaskExecutionMetad
 	o.SetLabels(utils.UnionMaps(o.GetLabels(), utils.CopyMap(taskCtx.GetLabels()), cfg.DefaultLabels))
 	o.SetName(taskCtx.GetTaskExecutionID().GetGeneratedName())
 
-	if !e.disableInjectOwnerReferences {
+	if !e.plugin.GetProperties().DisableInjectOwnerReferences {
 		o.SetOwnerReferences([]metav1.OwnerReference{taskCtx.GetOwnerReference()})
 	}
 
-	if cfg.InjectFinalizer && !e.disableInjectFinalizer {
+	if cfg.InjectFinalizer && !e.plugin.GetProperties().DisableInjectFinalizer {
 		f := append(o.GetFinalizers(), finalizer)
 		o.SetFinalizers(f)
 	}
 }
 
 func (e *PluginManager) GetProperties() pluginsCore.PluginProperties {
-	return pluginsCore.PluginProperties{}
+	props := e.plugin.GetProperties()
```

Comment:
why not just return e.plugin.GetProperties()?

Reply:
right, so in this case `e.plugin` is a `k8s.Plugin` and `GetProperties` here returns a `k8s.PluginProperties` which can contain properties that only make sense for `k8s`.

Reply:
aah oops, sorry missed that